File: csgo-leagues-ios/csgo-leagues-ios/Data/Mappers/TeamMapper.swift
import Foundation

struct TeamMapper {
    static func toDomain(_ dto: TeamDTO) -> Team {
        Team(
            id: dto.id,
            name: dto.name,
            imageURL: dto.imageUrl.flatMap { URL(string: $0)?.pandaScoreThumbnail }
        )
    }
}

File: csgo-leagues-android/app/src/main/java/com/voidx/csgo/data/mapper/TeamMapper.kt
package com.voidx.csgo.data.mapper

import com.voidx.csgo.data.dto.TeamDTO
import com.voidx.csgo.domain.entity.Team

object TeamMapper {
    fun toDomain(dto: TeamDTO): Team =
        Team(
            id = dto.id,
            name = dto.name,
            imageUrl = dto.imageUrl.toPandaScoreThumbnailUrl(),
        )
}
